include/odp/api/spec/packet.h

```diff
@@ -123,20 +123,38 @@ void odp_packet_free_sp(const odp_packet_t pkt[], int num);
 /**
  * Reset packet
  *
- * Resets all packet metadata to their default values. Packet length is used
- * to initialize pointers and lengths. It must be less than the total buffer
- * length of the packet. Packet is not modified on failure.
+ * Resets all packet metadata and adjusts packet data to start according to pool
+ * parameters (align, seg_len, headroom). Packet data length is set to 'len'.
+ * Maximum value for 'len' can be queried with odp_packet_reset_max_len()
+ * function.
```

Comment:
To be clear, it's allowed to pass len >max_len (will return failure). Len <= max_len will succeed.